### DISCUSSION TITLE
Auto-instrumentation with JMX

### DISCUSSION THREAD

**Discussion by sarkisjad** [open-telemetry/opentelemetry-java-instrumentation#11138] [Q&A] [answered]
Hello, 
I am auto-instrumenting and application by attaching a javaagent that emits metrics through the [jmx-metrics](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/jmx-metrics/javaagent)
I was wondering if there was a way to play around with the resource attributes.

For instance, I am aware that jmx auto instrumentation generates resource attributes like `process.pid`, `process.runtime.name`, `process.runtime.version` etc.

My goal is to remove some resource attributes, rename some and add others.
Can I do that through the rules.yml file for example? Or, can I maybe add some properties when attaching the javaagent? 
Could you please advise? 

Thanks

**Comment by laurit** [comment, marked as answer]
Resource attributes aren't specific to jmx-metrics it is sent along with all the telemetry generated from the agent or the otel java sdk. You can get rid of the undesired attributes by disabling the corresponding resource provider, see https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure#resource-provider-spi and https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources Or you can filer out the keys that you don't want https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure#opentelemetry-resource

**Reply by sarkisjad** [reply]
Hello @laurit 
Thank you for your reply. 
I was wondering if there was a way to add resource attributes as opposed to just filtering out some. 
I know we can add them through the OTEL_RESOURCE_ATTRIBUTES env var, but I was wondering if there was another way.
Can we make use of some extensions for that perhaps? https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension

**Reply by trask** [reply]
> Can we make use of some extensions for that perhaps? https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension

check out https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/examples/extension/src/main/java/com/example/javaagent/DemoResourceProvider.java

**Reply by sarkisjad** [reply]
Thank you for your response 
One more question, what would be the maven dependency to make use of to get these classes 
https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources

**Reply by laurit** [reply]
https://central.sonatype.com/artifact/io.opentelemetry.instrumentation/opentelemetry-resources
